helpful.el

```diff
@@ -2972,7 +3005,13 @@ See also `helpful-max-buffers'."
 
   ;; Enable users to bookmark helpful buffers.
   (set (make-local-variable 'bookmark-make-record-function)
-       #'helpful--bookmark-make-record))
+       #'helpful--bookmark-make-record)
+
+  ;; This function should normally only be called once after Org and
+  ;; helpful are loaded. To avoid using `eval-after-load' (which is
+  ;; only recommended in user init files), the function is called each
+  ;; time the major mode is used.
+  (helpful--add-support-for-org-links))
```

Comment:
The registration is not ideal. Calling the registration on every mode invocation is worse than using `with-eval-after-load`. The recommendation shouldn't be followed always. There is an actual problem here - as long as no helpful buffer has been created, help links are not registered for helpful.

To fix this it is sufficient to perform the registration in an `with-eval-after-load 'org` block in helpful.el at the toplevel. If the user wants to use helpful directly for Org help links they should then add `(with-eval-after-load 'org (require 'helpful))` to their config. If loading helpful always with org is considered too heavy one could create a separate file helpful-ol, which doesn't directly load helpful. See my osm-ol package as an example: https://github.com/minad/osm/blob/main/osm-ol.el

Reply:
I'm fine implementing that if @Wilfred agrees.